Automattic/sensei
# Refactor lesson reorder page to use `Sensei_Course_Structure`

### Changes proposed in this Pull Request

* It basically refactors the listing and the saving for the lessons reorder.
* It also removes some logic that was used to add styles to the reordering pages. These styles were added through CSS files only.

We could also refactor more things here, like moving all the reorder logic of courses, modules, and lessons to a separate class, and add all logic there. Or we could move them to their specific classes because the logic to order courses and lessons are in the `Sensei_Admin` class, and the logic to order the modules is in the `Sensei_Core_Modules` class. But I focused only on refactor them to use the `Sensei_Course_Structure`. If someone thinks it worths invest time in a bigger refactor in all this code now, we can go ahead.

### Testing instructions

**Note:** You can review the code here, and test directly in the #3672 because this code was partially refactored there, so you can prevent retest the same thing twice.

In summary, it should work as currently.

* Create some courses (some with modules and lessons, some with only lessons, some without content).
* Go to WP-admin > Lessons > Order Lessons, and reorder the content from the created courses.
* Go to the course pages and make sure the content was reordered.


### Deprecated Code

* `Sensei_Admin::get_lesson_order`
* `Sensei_Lesson::on_lesson_published`
